### PR TITLE
Delete kubernetes resources when deleting instance

### PR DESCRIPTION
**Pull request by macat**

### CODE REVIEW

**Comment by cored** [comment]
:shipit:

File: deployment/manifest_step.go [hunk 1]
```diff
@@ -79,3 +80,21 @@ func (s *ManifestStep) Deploy() error {
...
+	if err != nil {
+		return err
+	}
+	switch oGVK.Kind {
+	case "ReplicationController":
+		client.ReplicationControllers(namespace).Delete(meta.Name, nil)
+	case "Service":
+		client.Services(namespace).Delete(meta.Name, nil)
+	case "Pod":
+		client.Pods(namespace).Delete(meta.Name, nil)
```
Comment (bobbytables) [inline deployment/manifest_step.go]: Can these fail at all? Are these unchecked errors?

File: deployment/kubernetes.go [hunk 2]
```diff
@@ -88,6 +89,20 @@ func (d *KubernetesDeployment) Deploy() error {
...
+// Destroy deletes Kubernetes resourses
+func (d *KubernetesDeployment) Destroy() error {
+	tasksteps, err := d.steps()
+	if err != nil {
+		return err
+	}
+
+	for i, taskstep := range tasksteps {
+		glog.Infof("%d. Deploying Task %s...", i, taskstep.task.Name)
+		taskstep.step.Destroy()
```
Comment (bobbytables) [inline deployment/kubernetes.go]: Unchecked error.

**Comment by macat** [comment]
@bobbytables fixed error handling

**Comment by bobbytables** [comment]
Dope.

File: deployment/manifest_step.go [hunk 3]
```diff
@@ -79,3 +80,25 @@ func (s *ManifestStep) Deploy() error {
...
+	case "ReplicationController":
+		err = client.ReplicationControllers(namespace).Delete(meta.Name, nil)
+	case "Service":
+		err = client.Services(namespace).Delete(meta.Name, nil)
+	case "Pod":
+		err = client.Pods(namespace).Delete(meta.Name, nil)
+	}
+	if err != nil {
+		return err
+	}
```
Comment (bobbytables) [inline deployment/manifest_step.go]: why not just return err? It could be nil :) @macat

Comment (macat) [inline deployment/manifest_step.go]: simplified.

**Pull request merged**